### ISSUE TITLE
[visualizer] Muscles attached to MovingPathPoints are not drawn

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#1104]
Can be reproduced by loading gait10dof18musc.osim (the vasti and rect_fem use MovingPathPoints).

**Comment by aymanhab** [comment]
Are these part of GeometryPath? And how about via points?

**Comment by aymanhab** [comment]
Also if using parent relationships are we guaranteed some ordering?

**Comment by aymanhab** [comment]
Also if using parent relationships are we guaranteed consistent ordering?

**Comment by tkuchida** [comment]
> Are these part of GeometryPath?

Yes, the points are listed under Millard2012EquilibriumMuscle/GeometryPath/PathPointSet in the model file.

> And how about via points?

`PathPoint`s and `ConditionalPathPoint`s show up, but `MovingPathPoint`s do not (nor do the line segments emanating from them).

> Also if using parent relationships are we guaranteed consistent ordering?

As far as I can tell, the points are listed in order in the model file ("gait10dof18musc.osim", Version="30000").

**Comment by tkuchida** [comment]
The issue is in GeometryPath.cpp: [this line](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/GeometryPath.cpp#L146) calls `getLocationInGround()` but it looks like [this line](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Model/GeometryPath.cpp#L156) was not updated. PR coming soon...

**Comment by chrisdembia** [comment]
@tkuchida based on the comments in that code I'm not sure that is correct:

```
    // Even though these are Points which are Components they are completey 
    // orphaned and not part of any system since they are populated during 
    // a simulation.
```

**Comment by tkuchida** [comment]
> based on the comments in that code I'm not sure that what you are thinking of doing will work:

I made the change and it works locally. Why would line 156 be different than line 146?

**Comment by chrisdembia** [comment]
> Why would line 156 be different than line 146?

Not sure. The comments seem to be inconsistent with line 146.

**Comment by tkuchida** [comment]
The comment looks like it's out of date; it says "Trying to getLocationInGround(state) will faile due to no underlying system." but `getLocationInGround(state)` is called on line 146.

**Comment by tkuchida** [comment]
Changing line 156 to
```cpp
pos = point->getLocationInGround(state);
```
appears to correct the problem, but there may be side effects of which I'm not aware. @aymanhab any ideas?

**Comment by aymanhab** [comment]
Not the proof of correctness can mostly be confirmed visually, so go for it
Tom.

**Comment by tkuchida** [comment]
Thanks @aymanhab. Do you think this is the right fix, @aseth1?

**Comment by tkuchida** [comment]
**TL;DR** - `getLocationInGround(state)` is, indeed, the correct method to use here.

The underlying issue is that MovingPathPoint derives from Station and therefore inherits the `location` property. Because a MovingPathPoint's location is a function of the state, MovingPathPoints do not use their `location` property and don't call `get_location()`; instead, they use a `getLocation(const SimTK::State& s)` method to compute their (changing) location in their parent Frame. In `GeometryPath::generateDecorations()`, line 156
```cpp
pos = point->getBody().getTransformInGround(state)*point->getLocation();
```
calls `get_location()` (via `getLocation()`), which returns the value of the `location` property—which works for PathPoints and ConditionalPathPoints but not MovingPathPoints. Replacing line 156 with
```cpp
pos = point->getLocationInGround(state);
```
works because `MovingPathPoint::getLocation(const SimTK::State& s)` gets called.
